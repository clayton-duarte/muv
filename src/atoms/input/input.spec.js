import React from 'react';

import Input from '.';

it('should render a input', () => {
  const wrapper = mount(
    <Input />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('should change its style on error', () => {
  const wrapper = mount(
    <Input error />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('should change its style on success', () => {
  const wrapper = mount(
    <Input success />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('should change its style if disabled', () => {
  const wrapper = mount(
    <Input disabled />,
  );
  expect(wrapper).toMatchSnapshot();
});

it('should change its style if required', () => {
  const wrapper = mount(
    <Input required />,
  );
  expect(wrapper).toMatchSnapshot();
});
