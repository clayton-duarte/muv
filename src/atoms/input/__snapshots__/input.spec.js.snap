// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change its style if disabled 1`] = `
<Input
  disabled={true}
  error={false}
  placeholder="placeholder"
  required={false}
  success={false}
  theme={
    Object {
      "color": Object {
        "black": "#484848",
        "gray": "#f0f0f0",
        "overlay": "rgba(0, 0, 0, .15)",
        "primary": "#0082c8",
        "secondary": "#00baf2",
        "success": "#33bb33",
        "warn": "#dd3333",
        "white": "#fefefe",
      },
      "shape": Object {
        "base": "1rem",
        "border": ".125rem solid rgba(0, 0, 0, .15)",
        "margin": ".25rem .125rem",
        "padding": ".825rem 1.2rem",
        "radius": ".25rem",
        "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
      },
      "text": Object {
        "color": "#484848",
        "size": "1rem",
      },
      "transition": Object {
        "time": ".2s ease",
      },
    }
  }
  type="text"
>
  <styled.input
    disabled={true}
    placeholder="placeholder"
    required={false}
    success={false}
    theme={
      Object {
        "color": Object {
          "black": "#484848",
          "gray": "#f0f0f0",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0082c8",
          "secondary": "#00baf2",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "shape": Object {
          "base": "1rem",
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "margin": ".25rem .125rem",
          "padding": ".825rem 1.2rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
        },
        "text": Object {
          "color": "#484848",
          "size": "1rem",
        },
        "transition": Object {
          "time": ".2s ease",
        },
      }
    }
    type="text"
    warn={false}
  >
    <input
      className="sc-bdVaJa joRTBB"
      disabled={true}
      placeholder="placeholder"
      required={false}
      type="text"
    />
  </styled.input>
</Input>
`;

exports[`should change its style if required 1`] = `
<Input
  disabled={false}
  error={false}
  placeholder="placeholder"
  required={true}
  success={false}
  theme={
    Object {
      "color": Object {
        "black": "#484848",
        "gray": "#f0f0f0",
        "overlay": "rgba(0, 0, 0, .15)",
        "primary": "#0082c8",
        "secondary": "#00baf2",
        "success": "#33bb33",
        "warn": "#dd3333",
        "white": "#fefefe",
      },
      "shape": Object {
        "base": "1rem",
        "border": ".125rem solid rgba(0, 0, 0, .15)",
        "margin": ".25rem .125rem",
        "padding": ".825rem 1.2rem",
        "radius": ".25rem",
        "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
      },
      "text": Object {
        "color": "#484848",
        "size": "1rem",
      },
      "transition": Object {
        "time": ".2s ease",
      },
    }
  }
  type="text"
>
  <styled.input
    disabled={false}
    placeholder="placeholder"
    required={true}
    success={false}
    theme={
      Object {
        "color": Object {
          "black": "#484848",
          "gray": "#f0f0f0",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0082c8",
          "secondary": "#00baf2",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "shape": Object {
          "base": "1rem",
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "margin": ".25rem .125rem",
          "padding": ".825rem 1.2rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
        },
        "text": Object {
          "color": "#484848",
          "size": "1rem",
        },
        "transition": Object {
          "time": ".2s ease",
        },
      }
    }
    type="text"
    warn={false}
  >
    <input
      className="sc-bdVaJa joRTBB"
      disabled={false}
      placeholder="placeholder"
      required={true}
      type="text"
    />
  </styled.input>
</Input>
`;

exports[`should change its style on error 1`] = `
<Input
  disabled={false}
  error={true}
  placeholder="placeholder"
  required={false}
  success={false}
  theme={
    Object {
      "color": Object {
        "black": "#484848",
        "gray": "#f0f0f0",
        "overlay": "rgba(0, 0, 0, .15)",
        "primary": "#0082c8",
        "secondary": "#00baf2",
        "success": "#33bb33",
        "warn": "#dd3333",
        "white": "#fefefe",
      },
      "shape": Object {
        "base": "1rem",
        "border": ".125rem solid rgba(0, 0, 0, .15)",
        "margin": ".25rem .125rem",
        "padding": ".825rem 1.2rem",
        "radius": ".25rem",
        "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
      },
      "text": Object {
        "color": "#484848",
        "size": "1rem",
      },
      "transition": Object {
        "time": ".2s ease",
      },
    }
  }
  type="text"
>
  <styled.input
    disabled={false}
    placeholder="placeholder"
    required={false}
    success={false}
    theme={
      Object {
        "color": Object {
          "black": "#484848",
          "gray": "#f0f0f0",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0082c8",
          "secondary": "#00baf2",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "shape": Object {
          "base": "1rem",
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "margin": ".25rem .125rem",
          "padding": ".825rem 1.2rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
        },
        "text": Object {
          "color": "#484848",
          "size": "1rem",
        },
        "transition": Object {
          "time": ".2s ease",
        },
      }
    }
    type="text"
    warn={true}
  >
    <input
      className="sc-bdVaJa gbJXtR"
      disabled={false}
      placeholder="placeholder"
      required={false}
      type="text"
    />
  </styled.input>
</Input>
`;

exports[`should change its style on success 1`] = `
<Input
  disabled={false}
  error={false}
  placeholder="placeholder"
  required={false}
  success={true}
  theme={
    Object {
      "color": Object {
        "black": "#484848",
        "gray": "#f0f0f0",
        "overlay": "rgba(0, 0, 0, .15)",
        "primary": "#0082c8",
        "secondary": "#00baf2",
        "success": "#33bb33",
        "warn": "#dd3333",
        "white": "#fefefe",
      },
      "shape": Object {
        "base": "1rem",
        "border": ".125rem solid rgba(0, 0, 0, .15)",
        "margin": ".25rem .125rem",
        "padding": ".825rem 1.2rem",
        "radius": ".25rem",
        "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
      },
      "text": Object {
        "color": "#484848",
        "size": "1rem",
      },
      "transition": Object {
        "time": ".2s ease",
      },
    }
  }
  type="text"
>
  <styled.input
    disabled={false}
    placeholder="placeholder"
    required={false}
    success={true}
    theme={
      Object {
        "color": Object {
          "black": "#484848",
          "gray": "#f0f0f0",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0082c8",
          "secondary": "#00baf2",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "shape": Object {
          "base": "1rem",
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "margin": ".25rem .125rem",
          "padding": ".825rem 1.2rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
        },
        "text": Object {
          "color": "#484848",
          "size": "1rem",
        },
        "transition": Object {
          "time": ".2s ease",
        },
      }
    }
    type="text"
    warn={false}
  >
    <input
      className="sc-bdVaJa gbhcEo"
      disabled={false}
      placeholder="placeholder"
      required={false}
      type="text"
    />
  </styled.input>
</Input>
`;

exports[`should render a input 1`] = `
<Input
  disabled={false}
  error={false}
  placeholder="placeholder"
  required={false}
  success={false}
  theme={
    Object {
      "color": Object {
        "black": "#484848",
        "gray": "#f0f0f0",
        "overlay": "rgba(0, 0, 0, .15)",
        "primary": "#0082c8",
        "secondary": "#00baf2",
        "success": "#33bb33",
        "warn": "#dd3333",
        "white": "#fefefe",
      },
      "shape": Object {
        "base": "1rem",
        "border": ".125rem solid rgba(0, 0, 0, .15)",
        "margin": ".25rem .125rem",
        "padding": ".825rem 1.2rem",
        "radius": ".25rem",
        "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
      },
      "text": Object {
        "color": "#484848",
        "size": "1rem",
      },
      "transition": Object {
        "time": ".2s ease",
      },
    }
  }
  type="text"
>
  <styled.input
    disabled={false}
    placeholder="placeholder"
    required={false}
    success={false}
    theme={
      Object {
        "color": Object {
          "black": "#484848",
          "gray": "#f0f0f0",
          "overlay": "rgba(0, 0, 0, .15)",
          "primary": "#0082c8",
          "secondary": "#00baf2",
          "success": "#33bb33",
          "warn": "#dd3333",
          "white": "#fefefe",
        },
        "shape": Object {
          "base": "1rem",
          "border": ".125rem solid rgba(0, 0, 0, .15)",
          "margin": ".25rem .125rem",
          "padding": ".825rem 1.2rem",
          "radius": ".25rem",
          "shadow": "0 .25rem .25rem 0 rgba(0, 0, 0, .15)",
        },
        "text": Object {
          "color": "#484848",
          "size": "1rem",
        },
        "transition": Object {
          "time": ".2s ease",
        },
      }
    }
    type="text"
    warn={false}
  >
    <input
      className="sc-bdVaJa joRTBB"
      disabled={false}
      placeholder="placeholder"
      required={false}
      type="text"
    />
  </styled.input>
</Input>
`;
